### ISSUE TITLE
Issue using instance methods of None

### ISSUE THREAD

**Issue by bijij** [microsoft/pyright#4562]
**Describe the bug**
Pyright fails to bind methods on `None`

**To Reproduce**
See the below code, pyright complains that an additional positional argument is expected.

**Expected behavior**
As `None` is an instance methods are bound and should have `self` implicitly passed.

**Screenshots or Code**
```python
None.__eq__(2)
```

**VS Code extension or command-line**
`pyright 1.1.292`

**Additional context**
Admittedly, this is a very niche bug, but I thought it might be worth bringing up regardless.

**Comment by umurdualima** [comment]
Ini tampak seperti pesan kesalahan yang muncul dari penggunaan Pyright,
yang merupakan alat linting untuk Python. Pesan kesalahan ini menunjukkan
bahwa Pyright gagal mengikat metode pada objek None. Ini mungkin terjadi
karena objek None tidak memiliki metode yang ditentukan oleh kode Anda.

Untuk memperbaiki masalah ini, pastikan bahwa Anda mengecek apakah objek
tersebut None sebelum memanggil metodenya. Misalnya:

pythonCopy code
if obj is not None:
    obj.method()


Pada tanggal Rab, 1 Feb 2023 pukul 19.59 Josh ***@***.***>
menulis:

**Comment by erictraut** [comment]
This is included in pyright 1.1.293, which I just published. It will also be included in a future release of pylance.
